compliance/collection/nodeinventorizer/caching_inventorizer.go

```diff
@@ -0,0 +1,342 @@
+		c.opts.BackoffWaitCallback(currentBackoff)
+	}
+	writeBackoff(calcNextBackoff(currentBackoff), c.opts.BackoffFilePath)
+
+	// if no inventory exists, or it is too old, collect a fresh one and save it to the cache
+	newInventory, err := collectInventory(nodeName) // opts.Scanner.Scan(opts.NodeName)
+	if err != nil {
+		return nil, err
+	}
+
+	err = writeInventory(newInventory, c.opts.InventoryCachePath)
+	if err != nil {
+		return nil, errors.Wrap(err, "persisting inventory to cache")
+	}
+
+	// Remove backoff directly before returning message, so that a failing/killed container does not lead to
+	// frequent rescans of a Node, which are costly and might impact Node performance
+	removeBackoff(c.opts.BackoffFilePath)
+	return newInventory, nil
+}
+
+// readBackoff returns a backoff if found in given file, or the MaxBackoff on any error.
+//
+// It also checks the loaded backoff to not exceed the configured MaxBackoff.
+func readBackoff(path string) time.Duration {
+	backoff := env.NodeScanInitialBackoff.DurationSetting()
+	maxBackoff := env.NodeScanMaxBackoff.DurationSetting()
+
+	backoffFileContents, err := os.ReadFile(path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			log.Debug("No node scan backoff file found, continuing without pause")
+			return backoff
+		}
+		log.Warnf("Error reading node scan backoff file, continuing with MaxBackoff of %v. Error: %v", maxBackoff, err)
+		return maxBackoff
+	}
+
+	// We have an existing backoff counter
+	backoff, err = time.ParseDuration(string(backoffFileContents))
+	if err != nil {
+		log.Warnf("Error reading node scan backoff duration from file, continuing with MaxBackoff of %v. Error: %v", maxBackoff, err)
+		return maxBackoff
+	}
+
+	// Sanity-check the backoff loaded from disk
+	if backoff > maxBackoff {
+		log.Warnf("Existing node scan backoff exceeds MaxBackoff. Continuing with MaxBackoff of %v", maxBackoff)
+		return maxBackoff
+	}
+
+	return backoff
+}
+
+func writeBackoff(backoff time.Duration, path string) {
+	err := os.WriteFile(path, []byte(backoff.String()), 0644)
+	if err != nil {
+		log.Warnf("Error writing node scan backoff file: %v", err)
+	}
+}
+
+func calcNextBackoff(currentBackoff time.Duration) time.Duration {
+	maxBackoff := env.NodeScanMaxBackoff.DurationSetting()
+	nextBackoffInterval := currentBackoff + env.NodeScanBackoffIncrement.DurationSetting()
+	if nextBackoffInterval > maxBackoff {
+		nextBackoffInterval = maxBackoff
+	}
+	return nextBackoffInterval
+}
+
+func removeBackoff(backoffFilePath string) {
+	if err := os.Remove(backoffFilePath); err != nil {
+		log.Warnf("Could not remove node scan backoff file, subsequent scans may be delayed: %v", err)
+	}
+}
+
+// inventoryWrap is a private struct that saves a given inventory alongside a creation timestamp which determines its freshness.
+type inventoryWrap struct {
+	Created   time.Time
+	Inventory string
+}
```

Comment:
Why not a 
```suggestion
type inventoryWrap struct {
	Created   time.Time
	Inventory *storage.NodeInventory
}
```
? 

This would make the marshallings easier as you would not need to convert `storage.NodeInventory` to JSON and then marshall the entire wrap to JSON again. I would allow you also to drastically simplify the read and write functions.